source/utils/Vector3i.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (C) 2011-2014  OpenDungeons Team
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef VECTOR3I_H_
+#define VECTOR3I_H_
+
+#include <cstdint>
+#include <iostream>
+#include <OgreVector3.h>
+
+const int mPrecisionDigits = 20;
+const int Unit = (1 << mPrecisionDigits);
```

Comment:
Both. But Unit is worst because so common that you may have conflicts when including other libraries or other class members. That's why it is a bad practice to use so common variable names especially in a header.

Concerning mPrecisionDigits, the problem is that you followed class member variable naming. That means that when someone will read your code, he will think mPrecisionDigits is a class member variable while it is a constant.